common/types/head.go

```diff
@@ -24,4 +26,7 @@ type Head[BLOCK_HASH Hashable] interface {
 	// HashAtHeight returns the hash of the block at the given height, if it is in the chain.
 	// If not in chain, returns the zero hash
 	HashAtHeight(blockNum int64) BLOCK_HASH
+
+	// Returns the total difficulty of the block
```

Comment:
nit: Also add comment that chains which don't use difficulty concept, will return 0

Reply:
done!